node/src/components/consensus/protocols/highway.rs

```diff
@@ -240,6 +249,21 @@ impl<I: NodeIdT, C: Context> HighwayProtocol<I, C> {
         now: Timestamp,
     ) -> Vec<CpResult<I, C>> {
         let start_time = Timestamp::now();
+        // Check whether we should change the round exponent.
+        // It's important to do it before the vertex is added to the state - this way if the last
+        // round has finished, we now have all the vertices from that round in the state, and no
+        // newer ones.
+        if vv.inner().value().is_some() {
+            // unwraps are safe, as if value is `Some`, this is already a vote
+            self.round_success_meter.new_proposal(
+                vv.inner().vote_hash().unwrap(),
+                vv.inner().timestamp().unwrap(),
+            );
```

Comment:
Or you can replace it with 
```rust
if let ValidVertex(Vertex::Vote(vote)) = vv {
  …
}
```

Reply:
I don't really like that, because that would couple this place in the code with a lot of internals of `ValidVertex`, `Vertex`, `SignedWireVote` and `WireVote`. Unwraps might not be pretty, but I think that's still a slightly better approach.

Reply:
But that's what it is – `ValidVertex(Vetex::Vote(…))`. `inner().unwrap()` is in my opinion unreadable – what is `inner`? what am I unwrapping from? Why do I have to unwrap etc.

Reply:
The problem here is, I'm not unwrapping `inner()` - I'm unwrapping the results of methods on `inner()`. I could do `let vertex = vv.inner();` and then do `vertex.vote_hash().unwrap()` etc. - do you think that would make it more readable?

And if I went one level lower, then I don't have the methods on `vote`, as they are defined on `vertex`. I could add them, of course, but I'm not sure if that's much better.

Reply:
I'd also prefer to limit the amount of implementation details we have to expose from `highway_core`. Not sure what's best in this specific case here, though.

Reply:
Ok, if you both prefer it this way. We already mention `ValidVertex` explicitly in the method and we have `vv.into()` which maps it into a `Vertex` file lines below so I don't think that would be a problem. We are already dealing with both of these types here anyway.